src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,165 @@
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("sort", "views")
+            .addQueryParameter("sort_type", "DESC")
+            .build()
+        return GET(url)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/manga-list.html".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("artist", "")
+            .addQueryParameter("author", "")
+            .addQueryParameter("group", "")
+            .addQueryParameter("m_status", "")
+            .addQueryParameter("name", query)
+            .addQueryParameter("genre", "")
+            .addQueryParameter("ungenre", "")
+            .addQueryParameter("magazine", "")
+            .addQueryParameter("sort", "last_update")
+            .addQueryParameter("sort_type", "DESC")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun mangaDetailsParse(response: Response): SManga = SManga.create().apply {
+        val doc = Jsoup.parse(response.body.string())
+        author = doc.select("ul.manga-info a[href^=\"manga-author\"]").joinToString { it.text() }
+        genre = doc.select("ul.manga-info a[href^=\"manga-list-genre\"]").joinToString { it.text() }
+        val statusText = statusRegex.find(doc.select(".manga-info li:has(i.fa-spinner) a").attr("href"))?.groupValues?.get(0) ?: ""
+        status = when (statusText) {
+            "on-going" -> {
+                SManga.ONGOING
+            }
+            "completed" -> {
+                SManga.COMPLETED
+            }
+            else -> {
+                SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val slug = urlRegex.find(manga.url)?.groupValues?.get(0) ?: ""
+        return GET("$baseUrl/app/manga/controllers/cont.Listchapterapi.php?slug=$slug")
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = Jsoup.parse(response.body.string())
+        val chapterList = doc.select("ul > a")
+        var lastNum = 0f
+        val chapters = chapterList.map { chapter ->
+            SChapter.create().apply {
+                name = chapter.attr("title").trim()
+                setUrlWithoutDomain(URL(URL(baseUrl), chapter.attr("href")).toString())
+                chapter_number = floatRegex.find(chapter.attr("title").trim())?.groupValues?.get(0)?.toFloat() ?: (lastNum + 0.01f)
+                lastNum = chapter_number
+            }
+        }
+        return chapters
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        return GET(getChapterUrl(chapter))
+    }
```

Comment:
The default implementation of `getChapterUrl()` is:
```kt
open fun getChapterUrl(chapter: SChapter): String {
    return pageListRequest(chapter).url.toString()
}
```
So you're gonna recursively call forever here. I don't understand.